src/Features/Core/Portable/MakeFieldReadonly/MakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,190 @@
+
+                compilationStartContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
+                {
+                    var isConstructor = operationBlockStartContext.OwningSymbol.IsConstructor();
+                    var isStaticConstructor = operationBlockStartContext.OwningSymbol.IsStaticConstructor();
+
+                    operationBlockStartContext.RegisterOperationAction(operationContext =>
+                    {
+                        var fieldReference = (IFieldReferenceOperation)operationContext.Operation;
+                        IFieldSymbol field = fieldReference.Field;
+
+                        (bool isCandidate, bool written) fieldState = TryGetOrComputeIsCandidateField(field, operationContext.Options, operationContext.CancellationToken);
+
+                        // Ignore fields that are not candidates or have already been analyzed to have been written outside the constructor/field initializer.
+                        if (!fieldState.isCandidate || fieldState.written)
+                        {
+                            return;
+                        }
+
+                        // Field writes: assignment, increment/decrement or field passed by ref.
+                        bool isFieldAssignemnt = fieldReference.Parent is IAssignmentOperation assignmentOperation && assignmentOperation.Target == fieldReference;
+                        if (isFieldAssignemnt ||
+                            fieldReference.Parent is IIncrementOrDecrementOperation ||
+                            fieldReference.Parent is IArgumentOperation argumentOperation && argumentOperation.Parameter.RefKind != RefKind.None)
+                        {
+                            // Reads and writes of instance fields inside constructor are ignored, except for the below cases:
+                            //  1. Instance reference of field being written is not the instance being initialized by the constructor.
+                            //  2. Field is being written inside a lambda or local function. 
+                            if (field.ContainingType == operationBlockStartContext.OwningSymbol.ContainingType &&
+                                (isConstructor && fieldReference.Instance?.Kind == OperationKind.InstanceReference && (!isFieldAssignemnt || fieldReference.Parent.Parent?.Kind != OperationKind.ObjectOrCollectionInitializer) ||
+                                 isStaticConstructor && field.IsStatic) &&
+                                !IsInAnonymousFunctionOrLocalFunction(fieldReference))
+                            {
+                                return;
+                            }
+
+                            OnFieldWrite(field);
+                        }
+                    }, OperationKind.FieldReference);
+                });
+
+                compilationStartContext.RegisterCompilationEndAction(compilationEndContext =>
+                {
+                    foreach (var kvp in fieldStateMap)
```

Comment:
you can deconstruct in the foreach variable.  i.e.   ```foreach (var (field, (isCandidate, written)) in ...)```

Reply:
> foreach (var (field, (isCandidate, written)) in ...)

That causes a syntax error.